### ISSUE TITLE
Panic on loading GLTF file with an odd number of Uint16 indices

### ISSUE THREAD

**Issue by ncarnahan** [bevyengine/bevy#156]
In `device_create_buffer()`, there is an assertion that the buffer description size is divisible by `wgt::COPY_BUFFER_ALIGNMENT` which is equal to 4. When uploading the index buffer for a GLTF file with an odd number of Uint16 indices, the size of the buffer will be divisible by 2, but not 4 and the program will panic.

```rust
    pub fn device_create_buffer<B: GfxBackend>(
        &self,
        device_id: id::DeviceId,
        desc: &wgt::BufferDescriptor<Label>,
        id_in: Input<G, id::BufferId>,
    ) -> id::BufferId {
        span!(_guard, INFO, "Device::create_buffer");

        let hub = B::hub(self);
        let mut token = Token::root();

        log::info!("Create buffer {:?} with ID {:?}", desc, id_in);

        if desc.mapped_at_creation {
            assert_eq!(
                desc.size % wgt::COPY_BUFFER_ALIGNMENT,
                0,
                "Buffers that are mapped at creation have to be aligned to COPY_BUFFER_ALIGNMENT"
            );
        }
```

It looks like this was fixed in wgpu by padding the buffer in `create_buffer_with_data()`.

https://github.com/gfx-rs/wgpu-rs/issues/449
https://github.com/gfx-rs/wgpu-rs/pull/462

It looks like wgpu is forked? I'm not sure what the process would be for updating it/moving fixes over.

**Comment by StarArawn** [comment]
Bevy uses a custom wgpu crate because wgpu has been changing quite rapidly lately. The custom wgpu crate should be the latest master from wgpu or relatively close to it. 

`create_buffer_with_data()` might not work in all situations in bevy. Instead we should likely ensure that buffers have the right alignment. I believe we can pad the data with empty bytes to make sure it is a multiple of 4.

**Comment by bonsairobo** [comment]
I think this commit takes care of the issue: https://github.com/bevyengine/bevy/pull/158/commits/c4f5372601c73764c6b6d6db70c43beb7495ddf4

**Comment by alice-i-cecile** [comment]
Closing as fixed <3

**Comment by bjorn3** [comment]
1d17ce9 is not part of any branch.

**Comment by bonsairobo** [comment]
Yup that PR never got merged.

**Comment by geckoxx** [comment]
It only solves the issue in `create_buffer`. In `create_buffer_with_data` it still panics at `device.create_buffer_init` because the data is not aligned.

**Comment by geckoxx** [comment]
I filed an issue in the wgpu repo gfx-rs/wgpu-rs#901 to clarify if it's a bug in bevy or wgpu.

**Comment by mockersf** [comment]
fixed in https://github.com/gfx-rs/wgpu-rs/pull/900 👍

**Comment by mockersf** [comment]
that was fast! https://crates.io/crates/wgpu/0.8.1 should fix it, could you check?

**Comment by geckoxx** [comment]
yes it is fixed with wgpu 0.8.1
